### ISSUE TITLE
PageRank: ARPACK and PRPACK results are different when self-loops are present

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1639]
**Describe the bug**
For a graph with self-loops, the ARPACK and PRPACK methods of `igraph_pagerank` return different results.

**To reproduce**
Illustration using the Mathematica interface:

```
g = Graph[{1 -> 2, 2 -> 3, 3 -> 1, 2 -> 2}]
```

<img width="142" alt="image" src="https://user-images.githubusercontent.com/1212871/106313008-d1721400-6267-11eb-9290-212e0a5f96a7.png">

```
In[16]:= IGPageRank[g, Method -> "PRPACK"]
Out[16]= {0.26592, 0.480056, 0.254024}

In[17]:= IGPageRank[g, Method -> "Arnoldi"]
Out[17]= {0.229172, 0.455622, 0.315206}
```

**Version information**
Latest `develop` as of writing.

**Comment by szhorvat** [comment]
A direct verification based on the formula from Newman's book gives the same result as PRPACK, not as ARPACK.

**Comment by szhorvat** [comment]
Fixed by afb6eb023736172335de2004a6b45d0b3442f3dc
